src/modes/hip/utils.cpp

```diff
@@ -59,7 +59,7 @@ namespace occa {
                      hipGetDeviceProperties(&hipProps, deviceId));
 
       if (hipProps.gcnArch) {
-        return "gfx" + toString(hipProps.gcnArch);
+        return hipProps.gcnArchName;
```

Comment:
@dmcdougall Thanks for adding this, sorry for the delay in checking PRs! Can we initialize `gcnArchName` to some constant value like 
```cpp
static const std::string UNSET_VALUE = "OCCA_UNSET_VALUE";
...
  // Initialize gncArchName
  memcpy(gcnArchName, UNSET_VALUE.c_str(), UNSET_VALUE.size() + 1);
...
  // Leverage the arch name if possible (version > 3.5)
  if (strcmp(gcnArchName, UNSET_VALUE.c_str())) {
    return hipProps.gcnArchName;
  } else {
    return "gfx" + toString(hipProps.gcnArch);
  }
```

Reply:
I accomplished the same thing with a version check.  Is this acceptable?

Reply:
```cpp
      if (hipProps.gcnArch) { // AMD or NVIDIA
#if HIP_VERSION >= 306
        return hipProps.gcnArchName;
#else
        return "gfx" + toString(hipProps.gcnArch);
#endif
      }
```
@dmcdougall Do we need to check that `gcnArchName` is set or will it always be set?

Reply:
Looks like it's safe from the tensor flow backend, good enough for me :)
![image](https://user-images.githubusercontent.com/1812355/104392416-03694380-5510-11eb-8bbb-9be673d23266.png)

Reply:
If it's not set, it's a ROCm bug, not an OCCA bug.  It will be set for new enough ROCm versions.